DLT File Splitting Settings

Renu-Priya411 opened the pull request:
The given dlt file can be split into smaller sizes and can be saved in local as per user's convenience

Signed-off by : Renu Priya Krishnamoorthy <RenuPriya.KA.Krishnamoorthy@bti.bmwgroup.com>

Renu-Priya411 committed:
File Splitting Settings has been implemented

The given dlt file can be split into smaller sizes and can be saved in local as per user's convenience

Signed-off by : Renu Priya Krishnamoorthy <RenuPriya.KA.Krishnamoorthy@bti.bmwgroup.com>

alexmucde commented:
- Split files only at the end of a DLT message
- Implement not in Mainwindow.cpp, is getting too large

Renu-Priya411 committed:
Splitting the DLT File wrt DLT Messages

File spliting into smaller sizes using DLT message. Logic removed from mainwindow and written to a new .cpp and .h file

Signed-off by : Renu Priya Krishnamoorthy <RenuPriya.KA.Krishnamoorthy@bti.bmwgroup.com>

shubhamshahaBMW commented on src/filespliting.cpp:
```diff
@@ -0,0 +1,203 @@
+        }
+    }
+
+    m_file->close();
```
when trying the feature, getting error "Unexpected EOF: file ends before message fully read." in application output. 

Also, last few logs (20-100 lines) are missed. Data loss is seen when splitting the files.
Please test and correct the behaviour.

Renu-Priya411 replied:
Data loss is calculated by comparing the size of the original DLT file with the total size of the generated split files. The difference between these two values is reported as lost bytes.[split_loss_stats.csv](https://jira.cc.bmwgroup.net/secure/attachment/12130171/12130171_split_loss_stats.csv) 

This indicates how many bytes from the source file did not make it into the split output, typically due to early parser termination, truncated input, or a write failure.

Separate tail-drop value to identify whether the split stopped at an incomplete message near the end of the file is also tracked.

PFA for reference
[split_loss_stats.csv](https://github.com/user-attachments/files/28629377/split_loss_stats.csv)

shubhamshahaBMW commented on src/filespliting.cpp:
```diff
@@ -0,0 +1,203 @@
+        QString sizeUnit = unitCombo->currentText();
+
+        qint64 multiplier = 1;
+        if (sizeUnit == "KB") multiplier = 1024LL;
```
I don't think KB split is really needed.
@mfukar  what do you think?

mfukar replied:
Yeah perhaps it's not necessary at this point.

Renu-Priya411 replied:
KB Split removed

Renu-Priya411 commented on src/filespliting.cpp:
```diff
@@ -0,0 +1,203 @@
+           // Reset to start
+    m_file->seek(0);
+
+    while (!m_file->atEnd()) {
```
Progress bar implemented

Renu-Priya411 commented on src/filespliting.cpp:
```diff
@@ -0,0 +1,203 @@
+        }
+
+
+        if (folderPath.isEmpty()) {
```
redundant checks removed

Renu-Priya411 committed:
Fix the data loss while splitting, Bugs during Live Logging

Fix the data loss while splitting, Bugs during Live Logging

shubhamshahaBMW commented on src/mainwindow.ui:
```diff
@@ -440,6 +441,73 @@
+          </layout>
+         </item>
+         <item>
+          <widget class="QTreeView" name="exploreView">
```
<img width="272" height="211" alt="Image" src="https://github.com/user-attachments/assets/2f9f2c89-98c6-4bd8-bc9d-34cfc6fa97ed" />

The widget is appearing. some UI issue.

Renu-Priya411 replied:
Issue resolved

shubhamshahaBMW commented on src/filespliting.cpp:
```diff
@@ -0,0 +1,245 @@
+//The size is casted toqint64 and them parsed to splitOutputFile function.
+//The destination path can also be selected by user which will also be parsend to splitOutFile function.
+
+void FileSpliting::splitDLTFile_triggered(QFile &file,QStringList path){
```
unused parameter QFile &file

Renu-Priya411 replied:
parameter removed.
